src/robot/run.py

```diff
@@ -442,6 +444,7 @@ def main(self, datasources, **options):
             suite.visit(ModelModifier(settings.pre_run_modifiers,
                                       settings.run_empty_suite, LOGGER))
         with pyloggingconf.robot_handler_enabled(settings.log_level):
+            text.set_max_error_lines(settings.max_error_lines)
             result = suite.run(settings)
             LOGGER.info("Tests execution ended. Statistics:\n%s"
                         % result.suite.stat_message)
```

Comment:
As commented below, `set_max_error_lines` isn't really needed because the code here can just set `text.MAX_ERROR_LINES = settings.max_error_lines`. The code here should store the old value and reset if after execution.